pkg/detector/detector.go

```diff
@@ -794,7 +794,17 @@ func (d *ResourceDetector) OnPropagationPolicyAdd(obj interface{}) {
 
 // OnPropagationPolicyUpdate handles object update event and push the object to queue.
 func (d *ResourceDetector) OnPropagationPolicyUpdate(oldObj, newObj interface{}) {
-	// currently do nothing, since a policy's resource selector can not be updated.
+	if resourceSelectorsDeepEqual(oldObj, newObj) {
+		return
+	}
+
+	key, err := ClusterWideKeyFunc(newObj)
+	if err != nil {
+		return
+	}
+
+	klog.V(2).Infof("Update PropagationPolicy(%s) resourceSelectors", key)
+	d.policyReconcileWorker.Add(key)
```

Comment:
It seems like we only add this pp into reconcile queue. However, the object which is selected by this pp but not matched yet after updating will not be reconciled right? `ReconcilePropagationPolicy` only adds all matching objects into `Processor`. I guess the old object should also be handled.

Reply:
> However, the object which is selected by this pp but not matched yet after updating will not be reconciled right? 

No. If the update adds a new `ResourceSelector` which aims to match a new resource, and if the resource exists in `waiting list` it will be [put into reconciling queue](https://github.com/karmada-io/karmada/blob/c0349fe42931e9de3eb18671b6c028750660a0be/pkg/detector/detector.go#L779-L793).

> ReconcilePropagationPolicy only adds all matching objects into Processor. I guess the old object should also be handled.

Why do you think the old object should be handled? I guess the `handled` means `re-concile`. 
This reminds me that if an old object is matched by `previous PP`, but after the update, it doesn't match the `new PP`, it equals to remove the PP, we should clear the labels from the old object, just like https://github.com/karmada-io/karmada/blob/c0349fe42931e9de3eb18671b6c028750660a0be/pkg/detector/detector.go#L915-L949

Reply:
When there is an object which is selected by this pp but not matched yet after updating, I think we can ignore it. The life cycle of the rb is consistent with the life cycle of the resource,  and it will be deleted as the resource is deleted.

Therefore, in general, when modifying the `ResourceSelector` of the pp, we encourage the users to add some items rather than delete the old items at will. In this way, although some flexibility is lost, the behavior is more secure. In my opinion, the action of deleting the old item can be replaced by deleting the entire pp.

Reply:
It makes sense, I agree with you.

Reply:
>  I think the label in RBs will not be removed, which is not expected.

Hi @Garrybest , as you said, we need to remove the labels in the old RBs, otherwise, changes to the PP placement will still take effect on the old PRs.